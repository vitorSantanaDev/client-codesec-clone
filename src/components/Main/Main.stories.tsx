import { ComponentStory, ComponentMeta } from '@storybook/react'
import Main from './Main'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'title default',
    description: 'description default'
  }
} as ComponentMeta<typeof Main>

export const Basic: ComponentStory<typeof Main> = (args) => <Main {...args} />

Basic.args = {
  title: 'title basic',
  description: 'description basic'
}

export const Default: ComponentStory<typeof Main> = (args) => <Main {...args} />
