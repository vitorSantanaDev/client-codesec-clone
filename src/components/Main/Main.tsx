import { Title, Wrapper, Description, Logo, Illustration } from './styles'

const Main = ({
  title = 'React Avançado',
  description = 'TypeScript, ReactJS, NextJS e Styled Components'
}) => (
  <Wrapper>
    <Logo
      src="/images/logo.svg"
      alt="Imagem de um átomo rosa e 'React Avançado' escrito ao lado."
    />
    <Title>{title}</Title>
    <Description>{description}</Description>
    <Illustration
      src="/images/hero-illustration.svg"
      alt="Ilustração de um desenvolvedor sentado de frente para uma tela com código."
    />
  </Wrapper>
)

export default Main
