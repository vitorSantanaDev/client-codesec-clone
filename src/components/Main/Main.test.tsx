import { render, screen } from '@testing-library/react'
import Main from './Main'

describe('<Main />', () => {
  it('should render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avanÃ§ado/i })
    ).toBeInTheDocument()
  })

  it('should render correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
