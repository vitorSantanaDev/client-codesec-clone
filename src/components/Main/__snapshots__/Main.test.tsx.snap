// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #fff;
  background-color: #06092b;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo rosa e 'React Avançado' escrito ao lado."
    class="c1"
    src="/images/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Ilustração de um desenvolvedor sentado de frente para uma tela com código."
    class="c4"
    src="/images/hero-illustration.svg"
  />
</main>
`
